src/hooks/api/useBulkOperationDetails.js

```diff
@@ -12,56 +12,62 @@ export const useBulkOperationDetails = ({
   id,
   interval = 0,
   additionalQueryKeys = [],
+  shouldRefetch = false,
   ...options
 }) => {
   const ky = useOkapiKy();
   const [namespaceKey] = useNamespace({ key: BULK_OPERATION_DETAILS_KEY });
   const history = useHistory();
-  const [refetchInterval, setRefetchInterval] = useState(interval);
+  const [refetchInterval, setRefetchInterval] = useState(0);
   const { showErrorMessage } = useErrorMessages();
 
-  const clearInterval = () => {
-    setRefetchInterval(0);
-  };
+  const stopRefetching = () => setRefetchInterval(0);
 
-  const clearIntervalAndRedirect = (pathname, searchParams) => {
-    clearInterval();
+  const fetchBulkOperationDetails = async () => {
+    try {
+      const response = await ky.get(`bulk-operations/${id}`).json();
+      showErrorMessage(response);
 
-    history.replace({
-      pathname,
-      search: searchParams ? buildSearch(searchParams, history.location.search) : '',
-    });
+      if (response.status === JOB_STATUSES.FAILED || response?.errorMessage) {
+        stopRefetching();
+      }
+
+      return response;
+    } catch (error) {
+      showErrorMessage(error);
+      stopRefetching();
+
+      return error;
+    }
   };
 
-  const { data, isLoading } = useQuery({
-    queryKey: [BULK_OPERATION_DETAILS_KEY, namespaceKey, id, refetchInterval, ...additionalQueryKeys],
+  const { data, isLoading, refetch } = useQuery({
+    queryKey: [BULK_OPERATION_DETAILS_KEY, namespaceKey, refetchInterval, id, ...additionalQueryKeys],
     enabled: !!id,
-    refetchInterval,
-    queryFn: async () => {
-      try {
-        const response = await ky.get(`bulk-operations/${id}`).json();
+    queryFn: fetchBulkOperationDetails,
+    ...options,
+  });
 
-        showErrorMessage(response);
+  useEffect(() => {
+    if (!shouldRefetch || interval <= 0) return;
```

Comment:
I don't think you can do this and in come cases return callback for clean up... you even have a linter error